remove-groups/src/Filter.h

```diff
@@ -0,0 +1,111 @@
+#ifndef REMOVE_WATERLAYER_FILTER_H
+#define REMOVE_WATERLAYER_FILTER_H
+
+#include <cassert>
+#include "Reader.h"
+
+void removeGroups(Mesh &mesh,
+                  const std::set<unsigned int> &groupsToRemove) {
+  int removedCells = 0;
+  // Map old element/vertex id -> new element/vertex id
+  std::unordered_map<int, int> elementMap;
+  std::unordered_map<int, int> vertexMap;
+
+  std::cout << "Finding cells to remove" << std::endl;
+  auto elementCounter = 0UL;
+  auto vertexCounter = 0UL;
+  for (auto cell = 0u; cell < mesh.elementSize; ++cell) {
+    if (groupsToRemove.count(mesh.elementGroups[cell]) > 0) {
+      // Remove cell
+      ++removedCells;
+      mesh.elementGroups[cell] = 42;
+      const auto &neighbors = mesh.neighbors[cell];
```

Comment:
consider adding following comment:
for each neighbor cell of the current cell, update the component of the boundary array associated with the current cell to 1 (free surface).